relay-profiling/src/lib.rs

```diff
@@ -133,15 +137,22 @@ fn minimal_profile_from_json(data: &[u8]) -> Result<MinimalProfile, ProfileError
     serde_json::from_slice(data).map_err(ProfileError::InvalidJson)
 }
 
-pub fn expand_profile(payload: &[u8]) -> Result<Vec<u8>, ProfileError> {
-    let profile: MinimalProfile = minimal_profile_from_json(payload)?;
-    match profile.version {
+pub fn expand_profile(payload: &[u8]) -> Result<(EventId, Vec<u8>), ProfileError> {
+    let profile: MinimalProfile = match minimal_profile_from_json(payload) {
```

Comment:
The compiler can infer this type, no type hint needed.

```suggestion
    let profile = match minimal_profile_from_json(payload) {
```